### PR TITLE
Admins no longer auto-deadmin when logging in to a pAI

### PR DESCRIPTION
**Pull request by Ivniinvi**
## About The Pull Request
see title

## Why It's Good For The Game
pAIs are exempt from deadmin regulations, and it's really annoying to press readmin every single time you aghost (and potentially miss logs in that interval)

## Changelog
:cl:
admin: You no longer auto-deadmin as pAI
/:cl:

### CODE REVIEW

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Bad coding practice. Add a flag for PAIs or something broader that contains them. Don't add a hyper specific override.

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Still doesn't address the problem. The point of it being a flag is so that it can be disabled in config if we want. The `does_not_deadmin` var still acts as a needless override for that.

File: code/modules/mob/living/silicon/pai/pai.dm [hunk 1]
```diff
@@ -19,6 +19,7 @@
 	head_icon = 'icons/mob/pai_item_head.dmi'
 	var/network = "ss13"
 	var/obj/machinery/camera/current = null
+	does_not_deadmin = TRUE
```
Comment (Crossedfall) [inline code/modules/mob/living/silicon/pai/pai.dm]:
```suggestion
```

File: code/modules/mob/living/silicon/silicon.dm [hunk 2]
```diff
@@ -23,6 +23,7 @@
 	var/radiomod = "" //Radio character used before state laws/arrivals announce to allow department transmissions, default, or none at all.
 	var/obj/item/camera/siliconcam/aicamera = null //photography
 	hud_possible = list(ANTAG_HUD, DIAG_STAT_HUD, DIAG_HUD, DIAG_TRACK_HUD)
+	var/does_not_deadmin = FALSE //does this silicon trigger the deadmin condition
```
Comment (Crossedfall) [inline code/modules/mob/living/silicon/silicon.dm]:
```suggestion
```

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Still have some left over bits of code.

**Comment by Ivniinvi** [comment]
arrrrrrrrgh i thought i removed those

File: code/modules/mob/living/silicon/login.dm [hunk 3]
```diff
@@ -10,6 +10,8 @@
 /mob/living/silicon/auto_deadmin_on_login()
 	if(!client?.holder)
 		return TRUE
-	if(CONFIG_GET(flag/auto_deadmin_silicons) || (client.prefs?.toggles & DEADMIN_POSITION_SILICON))
+	if(!istype(src, /mob/living/silicon/pai) && (CONFIG_GET(flag/auto_deadmin_silicons) || (client.prefs?.toggles & DEADMIN_POSITION_SILICON)))
+		return client.holder.auto_deadmin()
+	if(istype(src, /mob/living/silicon/pai) && (CONFIG_GET(flag/auto_deadmin_pai) || (client.prefs?.toggles & DEADMIN_POSITION_PAI)))
```
Comment (Crossedfall) [inline code/modules/mob/living/silicon/login.dm:15]:
```suggestion
	if(istype(src, /mob/living/silicon/pai) && (CONFIG_GET(flag/auto_deadmin_pai) || (client.prefs?.toggles & DEADMIN_POSITION_PAI)))
		return client.holder.auto_deadmin()
	if(CONFIG_GET(flag/auto_deadmin_silicons) || (client.prefs?.toggles & DEADMIN_POSITION_PAI)))
```

**Review by Crossedfall — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/silicon/login.dm (hunk 3 above)
Comment (Crossedfall) [reply code/modules/mob/living/silicon/login.dm:15]: tbh, this logic is still pretty yucky.  Two checks for the same return is kinda nasty.

**Comment by ike709** [comment]
@Crossedfall

**Comment by Ivniinvi** [comment]
Crossed has said that he wants the following conditions to be met for the logic:
- One level of indenting 
- One typecheck
- No identical returns
I may be dumb but the only way I can think to accomplish this is one giant if statement, which he has also said he doesn't want.

**Comment by ike709** [comment]
@Crossedfall

**Comment by ike709** [comment]
I don't really like having a config flag just for pAI autodeadmin tbh.

**Pull request closed without merging**